File: 1A/WindowsAppTestVB/WindowsAppTestVB/Form1.vb
Public Class Form1

    Private original_color As Color
    Private Sub button_hi_Click(sender As Object, e As EventArgs) Handles button_hi.Click
        Me.RichTextBox1.Text = "Pippo"

    End Sub

    Private Sub button_clear_Click(sender As Object, e As EventArgs) Handles button_clear.Click
        Me.RichTextBox1.Text = ""
    End Sub

    Private Sub RichTextBox1_MouseEnter(sender As Object, e As EventArgs) Handles RichTextBox1.MouseEnter
        original_color = Me.RichTextBox1.BackColor
        Me.RichTextBox1.BackColor = Color.Aquamarine
    End Sub

    Private Sub RichTextBox1_MouseLeave(sender As Object, e As EventArgs) Handles RichTextBox1.MouseLeave
        Me.RichTextBox1.BackColor = original_color
    End Sub
End Class

File: 1A/WindowsFormsAppTest/WindowsFormsAppTest/Form1.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Linq;
using System.Security.Cryptography.X509Certificates;
using System.Text;
using System.Threading.Tasks;
using System.Windows.Forms;

namespace WindowsFormsAppTest
{
    public partial class Form1 : Form
    {
        private Color originalColor;

        public Form1()
        {
            InitializeComponent();
            

        }

        private void button1_Click(object sender, EventArgs e)
        {
            richTextBox1.Text = "World";
        }

        private void Form1_Load(object sender, EventArgs e)
        {

        }


        private void button_clear_Click(object sender, EventArgs e)
        {
            this.richTextBox1.Clear();
        }

        private void richTextBox1_MouseEnter(object sender, EventArgs e)
        {
            originalColor = this.richTextBox1.BackColor;
            this.richTextBox1.BackColor = Color.Azure;
        }

        private void richTextBox1_MouseLeave(object sender, EventArgs e)
        {

            this.richTextBox1.BackColor = originalColor;
        }
    }
}
